core/ir/instr_api.h

```diff
@@ -2351,6 +2356,16 @@ bool
 instr_is_reg_spill_or_restore(void *drcontext, instr_t *instr, bool *tls OUT,
                               bool *spill OUT, reg_id_t *reg OUT, uint *offs OUT);
 
+DR_API
+/**
+ * Same as \p instr_is_reg_spill_or_restore but also considers spills/restores using
+ * mcontext base in given reg.
```

Comment:
Ditto: is this "dr_mcontext_t"; when would this happen; how would the user know what reg to pass?